### PR TITLE
[#1927] Increase minimum version required for Git

### LINKED ISSUE #1927: Increase minimum version required for Git
### What feature(s) would you like to see in RepoSense

The [minimum version of Git required for RepoSense to run is `2.14`](https://reposense.org/ug/generatingReports.html). However, there is an additional requirement of Git version `2.23` for [the `--find-previous-authors` flag](https://reposense.org/ug/cli.html#find-previous-authors-f).

### Is the feature request related to a problem?

Having this additional requirement adds complexity - the need to check for the Git version when running RepoSense to support 2 different scenarios:

- Git versions `2.14` to `2.22`
- Git versions `2.23` and above

At this point in time, it is unlikely for anyone to have such an old version of Git. [Some versions](https://en.wikipedia.org/wiki/Git) had their last patch release in 2021, which suggests that they may not be worked on anymore.

### If possible, describe the solution

I propose raising the Git version requirement for both users and developers to `2.23`. 

Additionally, I suggest getting rid of the `GitVersion` checker entirely and letting the program fail when trying to find previous authors with an incompatible Git version. 

### Additional context

This should be flagged as a breaking change. Any release incorporating this should be numbered RepoSense `3.x.x` (assuming the current major version is still 2)

### PR DESCRIPTION
**Pull request by jq1836**
Fixes #1927

### Proposed commit message

```
Currently, RepoSense requires a minimum Git version of 2.14, with an
additional requirement of version 2.23 for the --find-previous-authors
flag. Supporting Git version 2.14 while requiring Git version 2.23 for
certain features adds complexity in the form of a Git version checker
to disable features depending on Git version 2.23.

Let's increase the minimum Git version required from 2.14 to 2.23.
```

### Other information

***Changes:***
* Removed `GitVersion` class.
* Updated `UG` and `DG` to reflect the increase in Git version required.
* Removed tests related to Git version.
* Removed messages related to Git version.

***Notes:***
RepoSense will still run perfectly fine with Git version `2.14 - 2.22` provided that the user does not use the `--find-previous-authors` flag in the CLI or have the `Find Previous Authors` column in `repo-config.csv` set to `yes`. Doing otherwise would result in the program failing.

### CODE REVIEW

**Review by MarcusTXK — CHANGES_REQUESTED** [review]
Great job! I think this is very well done, nice that you updated the relevant documentation as well.

Just one minor nitpick is that you missed out deleting a now unused variable `GIT_VERSION_INSUFFICIENT_MESSAGE`, located in `java/reposense/ConfigSystemTest.java`. Once that is cleaned up, this will be good to merge.

**Comment by jq1836** [comment]
Thanks for the review. Have made the relevant changes and am ready for review again.

**Review by MarcusTXK — APPROVED** [review]
LGTM, nice job!

**Review by sikai00 — APPROVED** [review]

File: docs/ug/cli.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ Binary file formats, such as `jpg`, `png`,`exe`,`zip`, `rar`, `docx`, and `pptx`
 <box type="info" seamless>
 
 * This flag only works on **git `2.23`** or later.
-* If an earlier version of **git** is used, RepoSense can still run but this flag will be ignored.
+* If an earlier version of **git** is used, RepoSense can still run but only if this flag is not used.
 </box>
```
Comment (chan-j-d) [inline docs/ug/cli.md:91]: I believe we can remove this section since we are assuming version 2.23.

File: docs/ug/configFiles.md [hunk 2]
```diff
@@ -46,6 +46,10 @@ Given below are the details of the various config files used by RepoSense.
 | Ignore File Size Limit | Enter **`yes`** to ignore both the default file size limit and the file size limit possibly set by the user in `repo-config.csv`. |
 | Skip Ignored File Analysis | Enter **`yes`** to ignore analysis of files exceeding the file size limit entirely. If file analysis is skipped, all information about the file will be omitted from the generated report. This option can significantly improve report generation time. |
 
+<box type="info" seamless>
+The Find Previous Authors option only works on **git `2.23`** or later.
+</box>
+
```
Comment (chan-j-d) [inline docs/ug/configFiles.md:52]: Same comment with this part

**Review by chan-j-d — CHANGES_REQUESTED** [review]
Thanks for doing this up. Wanted to request some small changes regarding the specification of 2.23 in UG

File: docs/ug/cli.md (hunk 1 above)
Comment (jq1836) [reply docs/ug/cli.md:91]: Thanks for pointing this out, will make the changes accordingly.

**Review by chan-j-d — APPROVED** [review]
LGTM!

**Pull request merged**